tests/integration_tests/datasets/api_tests.py

```diff
@@ -2498,9 +2333,11 @@ def test_get_or_create_dataset_creates_table(self):
         self.assertEqual(table.template_params, '{"param": 1}')
 
         db.session.delete(table)
+        db.session.commit()
```

Comment:
See [here](https://stackoverflow.com/a/28747810), specifically it mentions, 

> SQLite locks the database when a write is made to it, such as when an UPDATE, INSERT or DELETE is sent. When using the ORM, these get sent on flush. The database will remain locked until there is a COMMIT or ROLLBACK.

hence the need for the commit to free the lock prior to dropping the actual table.